From 0c32e9e2ef484e50f13e67280a7bb7235477c4cf Mon Sep 17 00:00:00 2001
From: Fengwei Yin <fengwei.yin@intel.com>
Date: Sun, 3 Apr 2011 00:32:25 +0800
Subject: [PORT FROM MAIN] ext4_utils: Fix an issue in add_entry

BZ: 19813

If the *offset + rec_len is in block boundary, the *offset will be
larger than len. The app abort.

This patch add one more block if detect possible *offset + ren_len
on block boundary.

Change-Id: I838c70c26a2f97443d1028de545b971743903269
Orig-Change-Id: I317bdcbfb3b888d9048838d050d178384397ad76
Signed-off-by: Fengwei Yin <fengwei.yin@intel.com>
Signed-off-by: Chenyang Du <chenyang.du@intel.com>
Reviewed-on: http://android.intel.com:8080/79204
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 ext4_utils/contents.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ext4_utils/contents.c b/ext4_utils/contents.c
index 3abbdc3..d1c9634 100644
--- a/ext4_utils/contents.c
+++ b/ext4_utils/contents.c
@@ -48,7 +48,7 @@ static u32 dentry_size(u32 entries, struct dentry *dentries)
 
 	for (i = 0; i < entries; i++) {
 		dentry_len = 8 + ALIGN(strlen(dentries[i].filename), 4);
-		if (len % info.block_size + dentry_len > info.block_size)
+		if (len % info.block_size + dentry_len >= info.block_size)
 			len += info.block_size - (len % info.block_size);
 		len += dentry_len;
 	}
-- 
1.7.4.1

